lib/licensed/dependency.rb

```diff
@@ -68,24 +68,24 @@ def local_files
     # Resets all local project and license information
     def reset_license!
       @project = nil
-      @matched_project_file = nil
+      @project_license = nil
       self.delete("license")
       self.text = nil
     end
 
-    # Returns the Licensee::ProjectFile representing the matched_project_file
-    # or remote_license_file
-    def project_file
-      matched_project_file || remote_license_file
-    end
-
-    # Returns the Licensee::LicenseFile, Licensee::PackageManagerFile, or
-    # Licensee::ReadmeFile with a matched license, in that order or nil
-    # if no license file matched a known license
-    def matched_project_file
-      @matched_project_file ||= project.matched_files
-                                       .select { |f| f.license && !f.license.other? }
-                                       .first
+    # Finds and returns a license from the Licensee::FSProject for this dependency.
+    def project_license
+      @project_license ||= if project.license && !project.license.other?
+        # if Licensee has determined a project to have a specific license, use it!
+        project.license
+      elsif project.license_files.map(&:license).uniq.size > 1
+        # if there are multiple license types found from license files,
+        # return 'other'
+        Licensee::License.find("other")
+      else
+        # check other project files if we haven't yet found a license type
+        project.licenses.reject(&:other?).first
```

Comment:
If a project is dual licensed, it will pick the first? What if it's dual licensed as GPL and Apache, in that order? Perhaps we should look for a non-copyleft license first to avoid false positives when a project is dual licensed (e.g., "choose" to use Apache over GPL in that case).

Reply:
This was added to maintain legacy behavior where `licensed` previously chose the first non-other license found from `project.matched_files`.  If it's a good idea to intentionally change that behavior, 👍